editor/find_in_files.cpp

```diff
@@ -874,6 +879,20 @@ void FindInFilesPanel::_on_replace_all_clicked() {
 	emit_signal(SNAME(SIGNAL_FILES_MODIFIED), modified_files);
 }
 
+void FindInFilesPanel::_on_button_clicked(TreeItem *p_item, int p_column, int p_id, int p_mouse_button_index) {
+	const String file_path = p_item->get_text(0);
+
+	if (_result_items.find(p_item)) {
+		_result_items.erase(p_item);
+	}
+
+	if (_file_items.find(file_path)) {
+		_file_items.erase(file_path);
+	}
+
+	memdelete(p_item);
```

Comment:
The signal is emitted here:
https://github.com/godotengine/godot/blob/6916349697a4339216469e9bf5899b983d78db07/scene/gui/tree.cpp#L3723
`cache.click_item`, which is being freed here, is not used afterwards, but this might change. Freeing the item that triggered the signal (even if it was emitted by the Tree) is better to be avoided.

You can use `get_tree()->queue_delete()`. It takes any Object.